pkg/ingester/circuitbreaker.go

```diff
@@ -253,3 +255,61 @@ func (cb *circuitBreaker) recordResult(errs ...error) error {
 	cb.metrics.circuitBreakerResults.WithLabelValues(circuitBreakerResultSuccess).Inc()
 	return nil
 }
+
+type ingesterCircuitBreaker struct {
+	push *circuitBreaker
+	read *circuitBreaker
+}
+
+func newIngesterCircuitBreaker(pushCfg CircuitBreakerConfig, readCfg CircuitBreakerConfig, logger log.Logger, registerer prometheus.Registerer) *ingesterCircuitBreaker {
+	return &ingesterCircuitBreaker{
+		push: newCircuitBreaker(pushCfg, registerer, circuitBreakerPushRequestType, logger),
+		read: newCircuitBreaker(readCfg, registerer, circuitBreakerReadRequestType, logger),
+	}
+}
+
+func (cb *ingesterCircuitBreaker) activate() {
+	if cb == nil {
+		return
+	}
+	cb.push.activate()
+	cb.read.activate()
+}
+
+// tryPushAcquirePermit tries to acquire a permit to use the push circuit breaker and returns whether a permit was acquired.
+// If it was possible, tryPushAcquirePermit returns a function that should be called to release the acquired permit.
+// If it was not possible, the causing error is returned.
+func (cb *ingesterCircuitBreaker) tryPushAcquirePermit() (func(time.Duration, error), error) {
+	finish, err := cb.push.tryAcquirePermit()
```

Comment:
This entire method is basically `return cb.push.tryAcquirePermit()`, right?

Reply:
Exactly. I have change this.
Thank you